src/com/google/enterprise/adaptor/database/TupleReader.java

```diff
@@ -151,49 +154,130 @@ protected void emitRow(ResultSet resultSet, ContentHandler handler)
       if (resultSet.getObject(col) == null) {
         atts.addAttribute("", "ISNULL", "ISNULL", "NMTOKEN", "true");
         handler.startElement("", columnName, columnName, atts);
-        // TODO: Re-examine this need for flush.
-        outWriter.flush();
         handler.endElement("", columnName, columnName);
         continue;
       }
       switch (sqlType) {
         case Types.DATE:
           Date dateValue = resultSet.getDate(col);
-          handler.startElement("", columnName, columnName, atts);
-          outWriter.write(DATEFMT.format(dateValue));
+          if (dateValue != null) {
```

Comment:
So you're changing the null handling here. Are there any tests of it? What does the change look like in the output? Are you delaying the construction of the writer just because the null handling is more localized?